var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -322,6 +336,18 @@ def cmake_args(self):
                 'ON' if '+intrepid' in spec else 'OFF'),
             '-DTrilinos_ENABLE_Intrepid2=%s' % (
                 'ON' if '+intrepid2' in spec else 'OFF'),
+            '-DTrilinos_ENABLE_Phalanx=%s' % (
+                'ON' if '+phal' in spec else 'OFF'),
+            '-DTrilinos_ENABLE_MiniTensor=%s' % (
+                'ON' if '+minitensor' in spec else 'OFF'),
+            '-DTrilinos_ENABLE_Isorropia=%s' % (
+                'ON' if '+isorropia' in spec else 'OFF'),
+            '-DTrilinos_ENABLE_Tempus=%s' % (
+                'ON' if '+tempus' in spec else 'OFF'),
+            '-DTrilinos_ENABLE_Rythmos=%s' % (
+                'ON' if '+rythmos' in spec else 'OFF'),
+            '-DTrilinos_ENABLE_Teko=%s' % (
+                'ON' if '+teko' in spec else 'OFF'),
```

Comment:
Please add checks like `conflict` for configurations you know are incompatible so that they fail in the spack resolution phase and not during Trilinos configure/build. For example, `Teko` has a required dependency on many packages including `Tpetra`, `ML`, etc. Thus, it would be nice to have things like
```python
    conflicts('+piro', when='~tpetra')
```
I understand that this would be time consuming, but I don't see a better solution (would be nice if we could somehow do parts of spack's trilinos package by parsing Trilinos `Dependencies.cmake`).

Reply:
I'll add that dependency. TriBits has an effective dependency system in place, we will need to dig through Dependencies.cmake and duplicate the dependencies in the Spack file for the options that I added. Would you be open to helping me with that?

Reply:
@gahansen Would be glad to help.

Reply:
Thanks! Email is inbound...